pkg/source/ttnv3/config/config.go

```diff
@@ -56,61 +56,66 @@ func (c *serverConfig) anyFieldEmpty() error {
 	return nil
 }
 
-func New() (*Config, *pflag.FlagSet) {
+func New() (*Config, source.FlagSets) {
 	var (
-		config = &Config{}
-		flags  = &pflag.FlagSet{}
+		config  = &Config{}
+		devices = &pflag.FlagSet{}
+		shared  = &pflag.FlagSet{}
 	)
 
-	flags.StringVar(&config.AppID,
+	devices.StringVar(&config.AppID,
```

Comment:
The `--app-id` flag was removed from `application` command of `ttnv{2,3}` sources since there was no use case for it.
Example:
```bash
ttn-lw-migrate ttnv3 application --app-id 'my-app'
```
If the tool was called like this without an argument it would expect to receive Application IDs from `stdin` so the flag would be ignored and the tool would 'hang unexpectedly' (waiting).

Since https://github.com/TheThingsNetwork/lorawan-stack-migrate/commit/f33a68b28ac0b0bb2fd987b411689a6b1374b763 the argument always overrides the flag. This combination makes the flag redundant or even confusing for the `application` command in those sources.